File: lista de exercicios/questão 8/questao.py
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

class Empresa:
    def __init__(self, nome):
        self.nome = nome
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

    def exibir_empregados(self):
        print(f"Empregados da empresa {self.nome}:")
        for empregado in self.empregados:
            print(f"Nome: {empregado.nome}, Cargo: {empregado.cargo}, Salário: {empregado.salario}")

empresa = Empresa("TechCorp")
empregado1 = Empregado("Ana", "Desenvolvedora", 5000.0)
empregado2 = Empregado("Carlos", "Designer", 4000.0)

empresa.adicionar_empregado(empregado1)
empresa.adicionar_empregado(empregado2)

empresa.exibir_empregados()

File: lista de exercicios/questão 8/questao.go
package main

import "fmt"

type Empregado struct {
    Nome   string
    Cargo  string
    Salario float64
}

type Empresa struct {
    Nome      string
    Empregados []Empregado
}

func (e *Empresa) AdicionarEmpregado(empregado Empregado) {
    e.Empregados = append(e.Empregados, empregado)
}

func (e Empresa) ExibirEmpregados() {
    fmt.Printf("Empregados da empresa %s:\n", e.Nome)
    for _, empregado := range e.Empregados {
        fmt.Printf("Nome: %s, Cargo: %s, Salário: %.2f\n", empregado.Nome, empregado.Cargo, empregado.Salario)
    }
}

func main() {
    empresa := Empresa{Nome: "TechCorp"}
    empregado1 := Empregado{Nome: "Ana", Cargo: "Desenvolvedora", Salario: 5000.0}
    empregado2 := Empregado{Nome: "Carlos", Cargo: "Designer", Salario: 4000.0}

    empresa.AdicionarEmpregado(empregado1)
    empresa.AdicionarEmpregado(empregado2)

    empresa.ExibirEmpregados()
}
